src/main/java/io/cdap/plugin/batch/source/ftp/FTPBatchSource.java

```diff
@@ -171,19 +173,23 @@ public void validate(FailureCollector collector) {
           } catch (Exception e) {
             throw new IllegalArgumentException(String.format("Unable to parse url: %s %s", e.getMessage(), e));
           }
-          Configuration conf = new Configuration();
+          Job job = JobUtils.createInstance();
+          Configuration conf = job.getConfiguration();
           for (Map.Entry<String, String> entry : fsp.entrySet()) {
             conf.set(entry.getKey(), entry.getValue());
           }
           String protocol = urlInfo.toUri().getScheme();
           if (protocol.equals(SFTP_PROTOCOL)) {
             conf.set(FS_SFTP_IMPL, SFTP_FS_CLASS);
           }
-          FileSystem fs = FileSystem.newInstance(urlInfo.toUri(), conf);
+          FileSystem fs = JobUtils.applyWithExtraClassLoader(job, getClass().getClassLoader(),
+              f -> FileSystem.get(urlInfo.toUri(), conf));
           // TODO: Add setTimeout option in the future
           // https://cdap.atlassian.net/browse/PLUGIN-1181
           fs.getFileStatus(urlInfo);
         } catch (Exception e) {
+          //Log exception details as otherwise we loose it, and it's hard to debug
+          LOG.warn("Unable to connect with the given url", e);
```

Comment:
nit: why not logging as error?

Reply:
We do propagate it up as a validation failure. The logging is mostly to make debugging easier as validation failure looses original message and usually does not show the original stack trace